app/src/pages/review.tsx

```diff
@@ -1,180 +1,151 @@
-    }
-  })
+import Button from '@components/Button'
+import PageError from '@components/PageError'
+import ReviewSection from '@components/ReviewSection'
 
-  return <List i18nKeys={i18nKeys} />
-}
+import {
+  EligibilityScreenerBody,
+  EligibilityScreenerResponse,
+} from '@pages/api/eligibility-screener'
 
-export const formatEligibilityResponses = (
-  session: DefaultState
-): ReviewElementProps[] => {
-  return [
-    {
-      labelKey: 'Eligibility.residential',
-      children:
-        (session?.eligibility?.residential && (
-          <Trans i18nKey={session?.eligibility?.residential} />
-        )) ||
-        null,
-    },
-    {
-      labelKey: 'Eligibility.categorical',
-      children: formatCategoricalOrAdjunctive(categoryKeys, session),
-    },
-    {
-      labelKey: 'Eligibility.before',
-      children:
-        (session?.eligibility?.before && (
-          <Trans i18nKey={session?.eligibility?.before.replace(/[2]/g, '')} />
-        )) ||
-        null,
-    },
-    {
-      labelKey: 'Eligibility.programs',
-      children: formatCategoricalOrAdjunctive(programKeys, session),
-    },
-  ]
-}
+import type { EditablePage } from '@src/types'
+import { initialSessionData } from '@utils/sessionData'
 
-export const formatClinicResponses = (
-  session: DefaultState
-): ReviewElementProps[] => {
-  return [
-    {
-      labelKey: 'Review.clinicSelected',
-      children:
-        (session?.clinic && (
-          <ClinicInfo
-            name={session?.clinic.clinic}
-            streetAddress={session?.clinic.clinicAddress}
-            phone={session?.clinic.clinicTelephone}
-            isFormElement={false}
-          />
-        )) ||
-        null,
-    },
-  ]
+interface ReviewProps extends EditablePage {
+  baseUrl: string
 }
 
-export const formatContactResponses = (
-  session: DefaultState
-): ReviewElementProps[] => {
-  const contactResponses: ReviewElementProps[] = []
-  contactKeys.forEach((key: string) => {
-    contactResponses.push({
-      labelKey: `Contact.${key}`,
-      children:
-        (session?.contact[key as Contact] && (
-          <Trans i18nKey={session?.contact[key as Contact]} />
-        )) ||
-        null,
-    })
-  })
-  return contactResponses
-}
+const Review: NextPage<ReviewProps> = (props: ReviewProps) => {
+  const {
+    session,
+    setSession,
+    backRoute = '',
+    forwardRoute = '',
+    baseUrl,
+  } = props
+
+  // Using form to store all of the data in a component state
+  // resolves all hydration issues.
+  const [form, setForm] = useState(initialSessionData)
+  useEffect(() => {
+    setForm(session)
+  }, [session])
+
+  const [errorMessage, setErrorMessage] = useState('')
+
+  const { t } = useTranslation('common')
+
+  const router = useRouter()
 
-const Review: NextPage = () => {
-  const { session } = useAppContext()
+  const buildEligibilityArrays = (data: string[]) => {
+    return data.map((category) => t(`Eligibility.${category}`))
+  }
```

Comment:
Notes for reviewers: This is a little silly, but `session` stores the object keys (e.g. 'pregnant', 'foster'), but we want to send the human-friendly string to the api (e.g. "I'm pregnant", "I'm the guardian or foster parent of an infant or child under the age of 5 years old").